enhancements/microshift/microshift-apiserver-certs.md

```diff
@@ -0,0 +1,302 @@
+---
+title: microshift-apiserver-certs
+authors:
+  - "@pacevedom"
+reviewers:
+  - "@fzdarsky, MicroShift architect"
+  - "@ggiguash, MicroShift contributor"
+  - "@stlaz, Security specialist"
+  - "@pmtk, MicroShift contributor"
+  - "@copejon, MicroShift contributor"
+  - "@deads2k, OpenShift architect"
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - N/A
+creation-date: 2023-01-20
+last-updated: 2023-01-20
+tracking-link:
+  - https://issues.redhat.com/browse/USHIFT-716
+see-also:
+  - N/A
+replaces:
+  - N/A
+superseded-by:
+  - N/A
+---
+
+# MicroShift API server certificates
+## Summary
+This enhancement proposes a strategy to manage API server certificates in all
+kinds of environments, both externally (accessible through kubeconfig) and
+internally (pods reaching the API server).
+
+MicroShift has a wide variety of different use cases when it comes to API
+server connectivity, we can find environments where the cluster needs to be
+reachable with the IP, with the hostname, by some other name having DNS, or
+a combination of them.
+
+## Motivation
+MicroShift does not cover all requirements and use cases right now. It shares
+certificate handling with OCP, which is a bit different in terms of deployment.
+
+MicroShift does not have load balancers before the API server, and is required
+to be reachable and authenticate by using the IP, hostname, a DNS name, or a
+combination of those. In order to fulfill industrial use cases, we propose
+having a single certificate.
```

Comment:
It's a little redundant, but I think it's important to be clear about the limited scope of the change.

```suggestion
having a single API server certificate.
```